bigbluebutton-tests/playwright/connectionFailure/connectionFailure.spec.js

```diff
@@ -0,0 +1,53 @@
+const { test, devices } = require('@playwright/test');
+const { ScreenShare, MultiUserScreenShare } = require('../screenshare/screenshare');
+const { sleep } = require('../core/helpers');
+const e = require('../core/elements');
+const { getCurrentTCPSessions, killTCPSessions } = require('./util');
+const notificationsUtil = require('../notifications/util');
+const deepEqual = require('deep-equal');
+
+test.describe.parallel('Connection failure', () => {
+  // https://docs.bigbluebutton.org/2.6/release-tests.html#sharing-screen-in-full-screen-mode-automated
+  test('Screen sharer', async ({ browser, browserName, page }) => {
+    test.skip(browserName === 'firefox' && process.env.DISPLAY === undefined,
+              "Screenshare tests not able in Firefox browser without desktop");
+    const screenshare = new ScreenShare(browser, page);
+    await screenshare.init(true, true);
+    await screenshare.startSharing();
+
+    await killTCPSessions(await getCurrentTCPSessions());
+
+    // I'd like to do this:
+    // await notificationsUtil.checkNotificationText(screenshare, "Code 1101. Try sharing the screen again.");
+
+    // but that code only checks the first toast, and there's a bunch of toasts that show up
+    // on this test.  So instead I use xpath, like this: (should we do it this way in checkNotificationText?)
+    await screenshare.hasElement('//div[@data-test="toastSmallMsg"]/span[contains(text(), "Code 1101. Try sharing the screen again.")]');
+  });
+
+  test('Screen share viewer', async ({ browser, browserName, page, context }) => {
+    test.skip(browserName === 'firefox' && process.env.DISPLAY === undefined,
+              "Screenshare tests not able in Firefox browser without desktop");
+    const screenshare = new MultiUserScreenShare(browser, context);
+
+    await screenshare.initModPage(page);
+    await screenshare.startSharing(screenshare.modPage);
+
+    const tcpModeratorSessions = await getCurrentTCPSessions();
+    //console.log('tcpModeratorSessions', tcpModeratorSessions);
+
+    await screenshare.initUserPage(false);
+    await screenshare.userPage.joinMicrophone();
+    await screenshare.userPage.hasElement(e.screenShareVideo);
+
+    const tcpSessions = await getCurrentTCPSessions();
+    // Other comparisons, like == or the array includes method, don't do a deep comparison
+    // and will always return false since the two arrays contain different objects.
+    const tcpUserSessions = tcpSessions.filter(x => !tcpModeratorSessions.some(e => deepEqual(e,x)));
```

Comment:
What's the deep-equal's advantage that we are using here? (comparing to Node.js's `assert.deepEqual()`) Is it the speed?

Reply:
Reading some more about this, it looks like `assert.deepEqual` will throw an exception rather than just return true or false.  As you noted, and as [the doc page](https://www.npmjs.com/package/deep-equal) for `deep-equal` notes, `deep-equal` is faster than wrapping `assert.deepEqual` in a try/catch block.

I think we should add `deep-equal` as a new npm dependency (as the PR currently does) and use it just because it's simpler.  I use it in a list filter, so it needs to return true or false.

Reply:
Sounds good